data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditMessage.java

```diff
@@ -146,10 +178,12 @@ public boolean equals(final Object o) {
             return false;
         }
         AuditMessage that = (AuditMessage) o;
-        return serviceName.equals(that.serviceName) &&
-                userId.equals(that.userId) &&
-                resourceId.equals(that.resourceId) &&
-                context.equals(that.context) &&
+        return leafResourceId.equals(that.leafResourceId) &&
+                token.equals(that.token) &&
+                Objects.equals(userId, that.userId) &&
+                Objects.equals(resourceId, that.resourceId) &&
```

Comment:
This does not need the `Objects.equals()` as it's a String value
```suggestion
                resourceId.equals(that.resourceId) &&
```

Reply:
This is odd. This is the generated code for the equals and we have the  "token" being handled correctly but the "userId" and "resourceId" are not.